modules/auth/ldap/ldap.go

```diff
@@ -124,33 +125,28 @@ func (ls *Source) SearchEntry(name, passwd string, directBind bool) (string, str
 		log.Trace("LDAP will use BindDN.")
 
 		var found bool
-		userDN, found = ls.FindUserDN(name)
+		userDN, found = ls.findUserDN(l, name)
 		if !found {
 			return "", "", "", "", false, false
 		}
 	}
 
-	l, err := ldapDial(ls)
-	if err != nil {
-		log.Error(4, "LDAP Connect error (%s): %v", ls.Host, err)
-		ls.Enabled = false
-		return "", "", "", "", false, false
-	}
-	defer l.Close()
-
-	log.Trace("Binding with userDN: %s", userDN)
-	err = l.Bind(userDN, passwd)
-	if err != nil {
-		log.Debug("LDAP auth. failed for %s, reason: %v", userDN, err)
-		return "", "", "", "", false, false
+	if directBind || !ls.AttributesInBind {
```

Comment:
Because `bind` changes the authentication context in LDAP connection, if we bind user before looking up & fetching attributes then this lookup is done in user context, if we bind user after, then the fetch is done in BindDN context. Which is anyway the subject of the commit.

Anyway I'll try to improve readability of this code part and I'll update the PR.